src/code/z_bgcheck.c

```diff
@@ -4331,18 +4331,22 @@ s32 SurfaceType_IsIgnoredByProjectiles(CollisionContext* colCtx, CollisionPoly*
 }
 
 /**
- * SurfaceType Get Conveyor Surface Type
- * Return type 0 (CONVEYOR_WATER) if 'poly' is a surface that will only move player underwater
- * Return type 1 (CONVEYOR_FLOOR) if `poly` is a surface that must be stood on to move player
- */
-s32 SurfaceType_GetConveyorType(CollisionContext* colCtx, CollisionPoly* poly, s32 bgId) {
+ * Checks if poly is a floor conveyor
+ *
+ * A conveyor surface is enabled with non-zero speed.
+ * When enabled, the conveyor will exhibit two types of behaviour depending on the return value:
+ *
+ * If true, then it is a floor conveyor and will push player only while being stood on
+ * If false, then it is a water conveyor and will push player only while in water
+ */
+s32 SurfaceType_IsFloorConveyor(CollisionContext* colCtx, CollisionPoly* poly, s32 bgId) {
     u32 flags;
 
     if (BgCheck_GetCollisionHeader(colCtx, bgId) == NULL) {
-        return CONVEYOR_FLOOR;
+        return true;
     }
     if (poly == NULL) {
-        return CONVEYOR_WATER;
+        return false;
     }
     flags = poly->flags_vIB & 0x2000;
     return !!flags;
```

Comment:
There was a bit of a discussion about this, can't remember if it bled into discord or not but here's the basis of it:
https://github.com/zeldaret/oot/pull/1103#discussion_r780762531

Reply:
Hmm, I remember reading that, although it was months ago. I think it's just a weird function all round. I'd prefer `Ground` over `Floor` for this sort of thing though since `Floor` usually indicates inside.

Reply:
I feel `Floor` has a fairly specific meaning when it comes to collision, and should probably keep it consistent.

Reply:
True, I suppose there is the floor-wall-ceiling not quite trichotomy.